v2/mongodb-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/mongodb/templates/MongoDbToBigQuery.java

```diff
@@ -77,11 +74,37 @@ public static boolean run(Options options) {
                 .withCollection(options.getCollection()))
         .apply(
             "Transform to TableRow",
-            MapElements.via(
-                new SimpleFunction<Document, TableRow>() {
-                  @Override
-                  public TableRow apply(Document document) {
-                    return MongoDbUtils.generateTableRow(document, userOption);
+            ParDo.of(
+                new DoFn<Document, TableRow>() {
+
+                  @ProcessElement
+                  public void process(ProcessContext c) {
+                    TableRow row = new TableRow();
+                    Document document = c.element();
+                    if (userOption.equals("FLATTEN")) {
+                      document.forEach(
+                          (key, value) -> {
+                            String valueClass = value.getClass().getName();
+                            switch (valueClass) {
+                              case "java.lang.Double":
+                                row.set(key, value);
+                                break;
+                              case "java.util.Integer":
+                                row.set(key, value);
+                                break;
+                              default:
+                                row.set(key, value.toString());
+                            }
+                          });
+                    } else {
+                      DateTimeFormatter timeFormat =
+                          DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss.SSS");
```

Comment:
This will create a new instance of DateTimeFormatter for each element, resulting in low performance. I believe the DateTimeFormatter instance can be made static final as it's immutable and thread-safe.

Reply:
I have moved the mapping part to utils to a static function and made the DateTimeFormatter as static final.